docs/README.md

```diff
@@ -998,9 +998,80 @@ _Reference_
 - https://www.freecodecamp.org/news/javascript-new-features-es2020/
 
 
+## 19. Iterators 
 
-<br/> <br/>  <br/>
+Iterators provide mechanism to iterate over collections.
 
+Following values are iterables:
+- Arrays
+- Strings
+- Maps
+- Sets
+- DOM data structures (work in progress)
+
+Plain objects are not iterable!
+
+Iterators are Objects which uses iterator protocols.
+- After defining iterator next() method is used to iterate over the collection.
+
+```
+let numbers = new Sequence(1,2,3,4,5,6);
+let iterator = numbers[Symbol.iterator]();
+
+let result = iterator.next();
+
+while(!result.done) {
+  console.log(result.value);
+  result=iterator.next()
+}
+```
+
+What is the difference between loops and iterators?
+
```

Comment:
**difference between loops and iterators?**
- You can loop through the list using a for loop as many times as you want, but you can only loop through the list using an iterator once. To loop again using an iterator you will need to reinitialize the iterator
-